Lib/dataclasses.py

```diff
@@ -1018,6 +1022,15 @@ def _process_class(cls, init, repr, eq, order, unsafe_hash, frozen,
         # sees a real default value, not a Field.
         if isinstance(getattr(cls, f.name, None), Field):
             if f.default is MISSING:
+                # https://github.com/python/cpython/issues/89529
+                if f.default_factory is not MISSING and not has_init_method:
+                    raise ValueError(
```

Comment:
`TypeError` maybe?

Reply:
I've used a ValueError because a ValueError is raised by `field` when you specify both `default` and `default_factory` or by this function when `eq != True` when `order = True`. I felt that not having `__init__` when you use `field(default_factory=...)` should be considered a ValueError since the issue is more on the use of `default_factory` rather than the lack of `__init__` in the class itself.